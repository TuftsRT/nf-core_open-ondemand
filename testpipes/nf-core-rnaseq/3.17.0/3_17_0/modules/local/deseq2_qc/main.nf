process DESEQ2_QC {
    label "process_medium"

    // (Bio)conda packages have intentionally not been pinned to a specific version
    // This was to avoid the pipeline failing due to package conflicts whilst creating the environment when using -profile conda
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-8849acf39a43cdd6c839a369a74c0adc823e2f91:ab110436faf952a33575c64dd74615a84011450b-0' :
        'biocontainers/mulled-v2-8849acf39a43cdd6c839a369a74c0adc823e2f91:ab110436faf952a33575c64dd74615a84011450b-0' }"

    input:
    path counts
    path pca_header_multiqc
    path clustering_header_multiqc

    output:
    path "*.pdf"                , optional:true, emit: pdf
    path "*.RData"              , optional:true, emit: rdata
    path "*pca.vals.txt"        , optional:true, emit: pca_txt
    path "*pca.vals_mqc.tsv"    , optional:true, emit: pca_multiqc
    path "*sample.dists.txt"    , optional:true, emit: dists_txt
    path "*sample.dists_mqc.tsv", optional:true, emit: dists_multiqc
    path "*.log"                , optional:true, emit: log
    path "size_factors"         , optional:true, emit: size_factors
    path "versions.yml"         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args  = task.ext.args  ?: ''
    def args2 = task.ext.args2 ?: ''
    def label_lower = args2.toLowerCase()
    def label_upper = args2.toUpperCase()
    prefix = task.ext.prefix ?: "deseq2"
    """
    deseq2_qc.r \\
        --count_file $counts \\
        --outdir ./ \\
        --cores $task.cpus \\
        --outprefix $prefix \\
        $args

    if [ -f "R_sessionInfo.log" ]; then
        sed "s/deseq2_pca/${label_lower}_deseq2_pca/g" <$pca_header_multiqc >tmp.txt
        sed -i -e "s/DESeq2 PCA/${label_upper} DESeq2 PCA/g" tmp.txt
        cat tmp.txt *.pca.vals.txt > ${label_lower}.pca.vals_mqc.tsv

        sed "s/deseq2_clustering/${label_lower}_deseq2_clustering/g" <$clustering_header_multiqc >tmp.txt
        sed -i -e "s/DESeq2 sample/${label_upper} DESeq2 sample/g" tmp.txt
        cat tmp.txt *.sample.dists.txt > ${label_lower}.sample.dists_mqc.tsv
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        bioconductor-deseq2: \$(Rscript -e "library(DESeq2); cat(as.character(packageVersion('DESeq2')))")
    END_VERSIONS
    """

    stub:
    def args2 = task.ext.args2 ?: ''
    def label_lower = args2.toLowerCase()
    prefix = task.ext.prefix ?: "deseq2"
    """
    touch ${label_lower}.pca.vals_mqc.tsv
    touch ${label_lower}.sample.dists_mqc.tsv
    touch ${prefix}.dds.RData
    touch ${prefix}.pca.vals.txt
    touch ${prefix}.plots.pdf
    touch ${prefix}.sample.dists.txt
    touch R_sessionInfo.log

    mkdir size_factors
    touch size_factors/${prefix}.size_factors.RData
    for i in `head $counts -n 1 | cut -f3-`;
    do
        touch size_factors/\${i}.size_factors.RData
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        bioconductor-deseq2: \$(Rscript -e "library(DESeq2); cat(as.character(packageVersion('DESeq2')))")
    END_VERSIONS
    """
}
