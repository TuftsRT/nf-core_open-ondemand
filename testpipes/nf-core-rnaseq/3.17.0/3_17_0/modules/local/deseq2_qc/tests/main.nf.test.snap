{
    "parse count data correctly - stub": {
        "content": [
            {
                "0": [
                    "deseq2.plots.pdf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "deseq2.dds.RData:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "deseq2.pca.vals.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "3": [

                ],
                "4": [
                    "deseq2.sample.dists.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "5": [

                ],
                "6": [
                    "R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "7": [
                    [
                        "RAP1_IAA_30M_REP1.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "RAP1_UNINDUCED_REP1.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "RAP1_UNINDUCED_REP2.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "WT_REP1.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "WT_REP2.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "deseq2.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "8": [
                    "versions.yml:md5,60f8e4f473397a4b5d1a3dc490c93ec5"
                ],
                "dists_multiqc": [

                ],
                "dists_txt": [
                    "deseq2.sample.dists.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "log": [
                    "R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "pca_multiqc": [

                ],
                "pca_txt": [
                    "deseq2.pca.vals.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "pdf": [
                    "deseq2.plots.pdf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "rdata": [
                    "deseq2.dds.RData:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "size_factors": [
                    [
                        "RAP1_IAA_30M_REP1.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "RAP1_UNINDUCED_REP1.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "RAP1_UNINDUCED_REP2.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "WT_REP1.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "WT_REP2.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "deseq2.size_factors.RData:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,60f8e4f473397a4b5d1a3dc490c93ec5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T11:38:08.92794"
    },
    "parse count data correctly": {
        "content": [
            [

            ],
            [
                "deseq2.sample.dists.txt:md5,1616ce40eb58ee75640efed219c6bbec"
            ],
            [
                "R_sessionInfo.log:md5,fb0da0d7ad6994ed66a8e68348b19676"
            ],
            [

            ],
            [
                "deseq2.pca.vals.txt:md5,4cda8e32ec39326c1350e77b4b607aad"
            ],
            [
                "deseq2.dds.RData:md5,c7c1872e7566321d0c3f7ca56c463088"
            ],
            [
                [
                    "RAP1_IAA_30M_REP1.txt:md5,ce4ff0b74aeedc3dc12a764c3504c9df",
                    "RAP1_UNINDUCED_REP1.txt:md5,51c9b23be5e82670aceb14f7e58dd867",
                    "RAP1_UNINDUCED_REP2.txt:md5,2c56106abcd9d31980f672b6e2862ee8",
                    "WT_REP1.txt:md5,5e3fe5f7e9b7f300be43649a8b496352",
                    "WT_REP2.txt:md5,2ce093f2c2e92c1bde1cd3da91ac884c",
                    "deseq2.size_factors.RData:md5,51c6643a0f8a9edabbff8df0165451f2"
                ]
            ],
            [
                "versions.yml:md5,60f8e4f473397a4b5d1a3dc490c93ec5"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T11:21:20.349616"
    }
}
